### PR TITLE
Add provenance tracking missing for input files

### PR DESCRIPTION
**Pull request by Bultako**
This PR fixes issue https://github.com/cta-observatory/ctapipe_io_lst/issues/25 adding missing provenance tracking for input files opened in several parts of the code. It also fixes the duplicated presence of a ``version.py`` file, and minor formatting aspects like import sorting or typos.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from setuptools import setup, find_packages
 from os import path
-from version import get_version, update_release_version
+from ctapipe_io_lst.version import get_version, update_release_version
```
Comment (MaxNoe) [inline setup.py:3]: This breaks the installation in a fresh environment, please revert.

There was a reason, why the version code, unfortunately, had to be duplicated.

**Comment by Bultako** [comment]
Thanks @MaxNoe 
I've reverted to the first commit.
I'll try to fix the duplicated `version.py` file issue in another PR

**Comment by MaxNoe** [comment]
@Bultako There were some requirements that I couldn't fulfill with anything I tried. 

But I think, we should just use setuptools_scm, let's see if we can use that to tick all the boxes.
However, I fear that doing `pip install <url to github tar ball>` will always result in an unknown version unless we hardcode a fallback version somewhere.

**Comment by Bultako** [comment]
@MaxNoe 
why not doing as others do?  i.e. `setup.py` file in `ctapipe`, which is adding the root dir to the path, this would result in

```
sys.path.insert(0, os.path.join(os.path.dirname(__file__)))
import ctapipe_io_lst # noqa

ctapipe_io_lst.version.update_release_version()
version = ctapipe_io_lst.version.get_version(pep440=True)
```

moreover, we are using the same `version.py` file, do we have different requirements?
if you think we can give it a try in this way I can open a new PR with this and continue discussion there.

**Comment by MaxNoe** [comment]
Because ctapipe does not import any of its dependencies in ctapipe/__init__.py, but this package does. So importing it in the setup.py would break installation in a fresh environment.

What you could do, is add `dirname(__file__) / ctapipe_io_lst` to path and just import version.

**Comment by Bultako** [comment]
Ok, let's continue with this in PR #47

**Review by rlopezcoto — APPROVED** [review]

**Pull request merged**